app/code/Magento/Catalog/Model/Product/Option/Type/File.php

```diff
@@ -228,7 +228,10 @@ public function validateUserValue($values)
          * previous configuration with no newly uploaded file
          */
         $fileInfo = null;
-        if (isset($values[$option->getId()]) && is_array($values[$option->getId()])) {
+        if (isset($values[$option->getId()])) {
+            if (is_string($values[$option->getId()])) {
```

Comment:
The same approach is used at app/code/Magento/Catalog/Model/Product/Option/Type/Select.php:100